source/blog/2016-11-11-testing-ovirt-changes-without-cluster.md

```diff
@@ -0,0 +1,188 @@
+```
+
+And the last option instructs the engine to send the hostname with every request that goes to the host. This allows us to use a single vdsmfake instance to simulate multiple hosts at once:
+
+```
+psql $ENGINE_DB -c "UPDATE vdc_options set option_value = 'true' WHERE option_name = 'UseHostNameIdentifier';"
+```
+
+Starting up the engine and adding the fake hosts
+A properly deployed and configured engine can be now started using the following command:
+
+```
+$ "${PREFIX}/share/ovirt-engine/services/ovirt-engine/ovirt-engine.py" start
+```
+
+Once the service boots up, you can access it by going to [http://localhost:8080](http://localhost:8080) and logging in using the engine credentials you provided during the setup phase.
+
+
+
+![oVirt welcome screen](/images/vdsmfake/welcome.png)
+
+
+
+Our fake hosts can be started by first cloning the [source](git://gerrit.ovirt.org/ovirt-vdsmfake.git), entering the ovirt-vdsmfake directory and using maven to start the [jetty](http://www.eclipse.org/jetty/) container:
+
+```
+# git clone git://gerrit.ovirt.org/ovirt-vdsmfake.git
+# pushd ovirt-vdsmfake
+# mvn clean jetty:run
+```
+
+
+
+You can now add as many hosts as you need to the engine using any made up names and addresses (just suffix them with `.vdsm.fake`). The engine requires you to enter a password when adding a host, but any random string is fine there (it won't be used anyway).
+
+
+![adding a new host](/images/vdsmfake/new-host.png)
+
+
+
+You have to connect a storage domain to make the engine fully operational so go to the Storage tab and create a new storage domain using any random NFS path. The fake hosts will report a connection success without actually talking to any remote server and this will enable the Datacenter functionality.
+
+![adding a new storage domain](/images/vdsmfake/storage.png)
+
+
+## What to do next?
+
+
+The setup is now ready and you can use most of the functionality to test your changes or just play with the engine. Starting, stopping and migrating VMs will work (no real VM will exist though, all is faked by the host simulator) as will most of the other administrative flows.
+
+
+In case you update the source code, just stop the engine using Control-C, execute the make install-dev and engine-setup (only needed if database structure was updated) commands to redeploy the changed pieces. Then you can start the engine again and test your changes.
+
+
+
+
+
+
+I hope this tutorial was useful for you and I am attaching a script that compiles all the steps into one simple command [dev-build.sh](/download/dev-build.sh) that does almost everything (including the engine-setup) for you. The script is very simple, so please tweak the `ROOT` path in it first to point to your `$OVIRT_PREFIX` directory. It also supports couple of command line options like `--clean` and `--config`. The default password it generates for the administrator user is `letmein`.
```

Comment:
My suggestion: "I hope this tutorial was useful for you. To make the process even simpler, I have attached a simple script that compiles all the steps into one simple command [dev-build.sh](/download/dev-build.sh) that does almost everything (including the engine-setup) for you. Before using it, you will need to tweak the `ROOT` path in it to point to your `$OVIRT_PREFIX` directory. It also supports couple of command line options like `--clean` and `--config`. The default password it generates for the administrator user is `letmein`."

Since there aren't any comments explaining what the command line flags do in the script, I would recommend either adding them to the script or adding a brief explanation here to save the user the time of trying to figure out what they do.